src/Engine/EmitMSIL/Replication.cs

```diff
@@ -568,9 +568,12 @@ private static object ExecWithZeroRI(CLRFunctionEndPoint finalFep, List<CLRStack
             var marshaller = ProtoFFI.CLRDLLModule.GetMarshaler(runtimeCore);
             CLRStackValue dsRetValue = marshaller.Marshal(result, ProtoFFI.CLRObjectMarshaler.GetProtoCoreType(finalFep.ReturnType), runtimeCore);
 
-            if (!dsRetValue.IsExplicitCall)
+            //TODO this causes a test failure. (array reduction to var not allowed)
+            //this is always false, and the comment below does not make sense.
+
+            // An explicit call requires return coercion at the return instruction
+            if (!isExplicitCall)
```

Comment:
I don't believe this concept of something being an explicit vs. an implicit call is relevant to us in the new engine anymore and strongly feel that we should avoid using this flag. I remember trying to work with this idea long ago when I was working on DesignScript but not touching it since then.

Reply:
It appears that it was originally used to distinguish between regular function calls vs. internal function calls like GC.

Reply:
thanks for the info @aparajit-pratap - it turns out atleast for this problem identified here the real issue is that this function should return `var[]..[]` but that the FEP says it returns `var` - I think @pinzart90 will address it in his type system import work.

in the meantime, I've marked some of these tests failures, which should turn green after Tibi's work - added 2 graph perf cases (1 failing for the aforementioned reason), and re-enabled a few of them that started passing!

Reply:
I do think I could construct a failing test case for the IsExplicitIssue though using the FFITarget ... maybe I will give that a shot so we have it tracked and can discuss how it should really be solved.

Reply:
actually, I can't create a test case for it that doesn't also require returning `object`